### PR TITLE
Add new Pylance setting

### PR DESCRIPTION
**Pull request by luabud**
We have a new Pylance setting called `python.analysis.languageServerMode`. More about it here: https://github.com/microsoft/pylance-release?tab=readme-ov-file#settings-and-customization

### CODE REVIEW

File: docs/python/settings-reference.md [hunk 1]
```diff
@@ -68,6 +69,7 @@ The language server settings apply when `python.languageServer` is `Pylance` or
 | useLibraryCodeForTypes | true | Parses the source code for a package when a type stub is not found. Available values are `true` and `false`. |
 | autoImportCompletions | false | Controls the offering of auto imports in completions. Available values are `true` and `false`. |
 | importFormat | `absolute`| Defines the default format when auto importing modules. Accepted values are `absolute` or `relative`. |
+| aiCodeActions | `true` | Whether to enable specific AI-assisted code actions. Requires the [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) extension to be enabled. <br> Accepted value is an object with a code action as key and a boolean as value. <br> Available code actions to use as keys: <ul> <li> `implementAbstractClasses`: enables code action to implement methods of classes inherited from an abstract class, using AI suggestions from GitHub Copilot to populate the method body.  </li> </ul>  Usage example: ```{"implementAbstractClasses": true}```
```
Comment (ntrogh) [inline docs/python/settings-reference.md:72]: The styling for the default value (`true` vs true) is inconsistent with most other entries. The previous line also has this.

The closing `|` seems to be missing from this line.

Comment (luabud) [inline docs/python/settings-reference.md:72]:
```suggestion
| aiCodeActions | true | Whether to enable specific AI-assisted code actions. Requires the [GitHub Copilot Chat](https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-chat) extension to be enabled. <br> Accepted value is an object with a code action as key and a boolean as value. <br> Available code actions to use as keys: <ul> <li> `implementAbstractClasses`: enables code action to implement methods of classes inherited from an abstract class, using AI suggestions from GitHub Copilot to populate the method body.  </li> </ul>  Usage example: ```{"implementAbstractClasses": true}``` |
```

File: docs/python/settings-reference.md [hunk 2]
```diff
@@ -68,6 +69,7 @@ The language server settings apply when `python.languageServer` is `Pylance` or
 | useLibraryCodeForTypes | true | Parses the source code for a package when a type stub is not found. Available values are `true` and `false`. |
 | autoImportCompletions | false | Controls the offering of auto imports in completions. Available values are `true` and `false`. |
 | importFormat | `absolute`| Defines the default format when auto importing modules. Accepted values are `absolute` or `relative`. |
```
Comment (luabud) [inline docs/python/settings-reference.md:71]:
```suggestion
| importFormat | absolute | Defines the default format when auto importing modules. Accepted values are `absolute` or `relative`. |
```

**Review by ntrogh — APPROVED** [review]
LGTM!

**Pull request merged**